homeassistant/components/spotify/__init__.py

```diff
@@ -71,6 +94,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         DATA_SPOTIFY_SESSION: session,
     }
 
+    if set(session.token["scope"].split(" ")) != set(SPOTIFY_SCOPES):
+        _LOGGER.warning("The scopes have changed and require reauthentication.")
```

Comment:
Should we put this inside the same code path that triggers the auth flow? Or trigger the reauth here ? btw, we can test if a set is a subset with `<=` so that rolling back doesn't force a re-auth.

Reply:
We can ditch this message IMHO, as we now handle it in the UI.

Reply:
Done